src/DependencyInjection/Compiler/ExtensionCompilerPass.php

```diff
@@ -215,6 +235,30 @@ protected function hasTrait(\ReflectionClass $class, $traitName)
         return $this->hasTrait($parentClass, $traitName);
     }
 
+    /**
+     * @phpstan-param class-string $class
+     * @phpstan-param class-string $subject
+     */
+    private function isExtensionClass(string $type, string $class, string $subject): bool
```

Comment:
Taking a look about the responsibility of this method, maybe we could find a better name. It is not only checking if a class is of a given type (which could be simplified with `is_a($class, $subject, true)`), but this is checking also the inheritance type.
Additionally, I think we should rethink the argument names (and their order), since we are checking if a `$class` is an instance of `$type` with the expected `$inheritance`.
I'm proposing something like this:
```
isInstanceOf(string $class, string $type, string $inheritance)
```

Reply:
Change it to `isSubtypeOf`, because `instanceof` is a reserved keyword and could be confusing.

Reply:
>Change it to isSubtypeOf, because instanceof is a reserved keyword and could be confusing.

Good, but please, note that I proposed `isInstanceOf()`, not `instanceof()`.
By the other hand, I still think the main concept of this method hasn't the right approach, conceptually speaking.
I think we are checking if a **subject** is subtype of a given **class**, but it currently is expressed as if we are checking if a **class** is a subtype of a given **subject**.